src/ServiceInsight/Framework/Commands/CloseWindowCommand.cs

```diff
@@ -0,0 +1,33 @@
+﻿namespace ServiceInsight.Framework.Commands
+{
+    using System;
+    using System.Windows;
+    using System.Windows.Input;
+
+    public class CloseWindowCommand : ICommand
+    {
+        readonly Window target;
+
+        public CloseWindowCommand(Window target)
+        {
+            this.target = target;
+        }
+
+        public bool CanExecute(object parameter)
+        {
+            return target != null;
+        }
+
+        public void Execute(object parameter)
+        {
+            if (CanExecute(parameter))
+            {
+                target.Close();
+            }
+        }
+
+        #pragma warning disable CS0067
```

Comment:
Minor nit-pick, but the two pragma line should not be indented. An auto-format will remove the indents, so it's probably better to get rid of them now.